include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,690 @@
+        This means the comparison is performed with
+        percent-decoding applied to the current string.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_URL_DECL
+    int
+    compare(decoded_view other) const noexcept;
+
+    /** Returns a string with the decoded data
+    */
+    std::string
+    to_string() const
+    {
+        std::string r;
+        assign_to(r);
+        return r;
+    }
+
+    /** Assigns the decoded characters to a container
+
+        This function allows the caller to recycle
+        the input string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class MutableString>
+    MutableString&
+    assign_to(MutableString& s) const;
+
+    /** Appends the decoded characters to a container
+
+        This function allows the caller to recycle
+        the input string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class MutableString>
+    MutableString&
+    append_to(MutableString& s) const;
+
+    /** Return true if two strings are equal
+    */
+    friend
+    bool
+    operator==(
+        decoded_view s0,
+        decoded_view s1) noexcept
+    {
+        return s0.compare(s1) == 0;
+    }
+
+    /** Return true if two strings are unequal
+    */
+    friend
+    bool
+    operator!=(
+        decoded_view s0,
+        decoded_view s1) noexcept
+    {
+        return s0.compare(s1) != 0;
+    }
+
+    /** Return true if s0 is less than s1
+    */
+    friend
+    bool
+    operator<(
+        decoded_view s0,
+        decoded_view s1) noexcept
+    {
+        return s0.compare(s1) < 0;
+    }
+
+    /** Return true if s0 is less or equal to s1
+    */
+    friend
+    bool
+    operator<=(
+        decoded_view s0,
+        decoded_view s1) noexcept
+    {
+        return s0.compare(s1) <= 0;
+    }
+
+    /** Return true if s0 is greater than s1
+    */
+    friend
+    bool
+    operator>(
+        decoded_view s0,
+        decoded_view s1) noexcept
+    {
+        return s0.compare(s1) > 0;
+    }
+
+    /** Return true if s0 is greater or equal to s1
+    */
+    friend
+    bool
+    operator>=(
+        decoded_view s0,
+        decoded_view s1) noexcept
+    {
+        return s0.compare(s1) >= 0;
+    }
+
+    /** Return true if two strings are equal
+
+        @par Constraints
+
+        @code
+        std::is_convertible< String, string_view >
+        @endcode
+
+    */
+    template <class String>
+    friend
+#ifndef BOOST_URL_DOCS
+    typename std::enable_if<
+        std::is_convertible<String, string_view>::value &&
+        !std::is_same<typename std::decay<String>::type, decoded_view>::value,
```

Comment:
You added an additional constraint without updating the javadoc. Was this needed to fix a compile error?

Reply:
That was a compiler error on a few old compilers that can't identify the overload with `decoded_view` should be preferred over the overload with the template.

Reply:
Good, then we shouldn't mention it in the javadoc